users/vieting/experiments/switchboard/ctc/feat/network_helpers/specaug_param_helper.py

```diff
@@ -0,0 +1,65 @@
+
+from typing import Dict, List
+
+def generate_specaug_params(
+    num_epochs: int,
+    base_values: Dict[str, int],
+    schedules: Dict[str, Dict[int, float]],
+    growth_strategy: str
+) -> Dict[str, List[int]]:
+    """
+    Generate specaug parameters for each epoch in advance.
+
+    Args:
+        num_epochs (int): Total number of epochs.
+        base_values (Dict[str, int]): Base values for each parameter.
+        schedules (Dict[str, Dict[int, float]]): Schedules for each parameter.
+        growth_strategy (str): Strategy for mask growth over epochs.
+
+    Returns:
+        Dict[str, List[int]]: Pre-generated parameter values for each epoch.
+    """
+    params = {key: [] for key in base_values.keys()}
+
+    for epoch in range(num_epochs):
+        for key, base_value in base_values.items():
+            schedule = schedules[key]
```

Comment:
Maybe you should assert that 0 (or 1, depending on where you start) is among the keys?